libkatana_python_native/include/katana/python/ErrorHandling.h

```diff
@@ -4,30 +4,60 @@
 #include <katana/Result.h>
 #include <pybind11/pybind11.h>
 
-namespace pybind11 {
+namespace katana::python {
 namespace detail {
 
 /// Cast an object to python unless it's already a python object.
 template <typename T>
 pybind11::object
-cast_if_needed(T&& v, return_value_policy policy, handle parent) {
+cast_if_needed(
+    T&& v, pybind11::return_value_policy policy, pybind11::handle parent) {
   return pybind11::cast(std::move(v), policy, parent);
 }
 
 template <>
 inline pybind11::object
 cast_if_needed<pybind11::object>(
-    pybind11::object&& v, return_value_policy, handle) {
+    pybind11::object&& v, pybind11::return_value_policy, pybind11::handle) {
   return std::move(v);
 }
 
+/// Convert @p err to a python exception and raise it.
+/// Always throws pybind11::error_already_set.
+KATANA_EXPORT pybind11::handle RaiseResultException(
+    const katana::ErrorInfo& err);
+
+}  // namespace detail
+
+template <typename T>
+T
+PythonChecked(const katana::Result<T>& src) {
```

Comment:
Is this suppose replace KATNA_CHECKED i saw in places?

Reply:
It doesn't need to. Though I did think about doing that. I think we should stick to using `KATANA_CHECKED` when possible. But it's not possible in all cases since some functions must have specific return types such as init implemenations.